### PR TITLE
Update teacher section login type and edit flow

### PR DESCRIPTION
**Pull request by TurnerRiley**
## Summary
Clarified wording and made visual tweaks for teachers choosing login types for sections.

## Links
Spec: https://docs.google.com/document/d/1YaC8yPL4QUNshj0zXmCZJ1o-FcGYb1Dexv9J0-acHuM/edit#
(sans Section 6 whose changes can be found at https://github.com/code-dot-org/code-dot-org/pull/40370)

### Original look:
OldSectionYesThirdParty

### New look (both in the case that the user has no 3rd party providers (1st picture) and the case that they do (2nd picture):
CreateNewSectionNoThirdParty
CreateNewSectionYesThirdParty

## Testing story
I tested manually checking that the first 5 requirements on the spec worked for different window sizes and different scenarios (in the case of requirement 3).

For requirement 3, I've tested all combinations of a user having/not having a third party provider for students (e.g. Google Classrooms) to ensure the third party login options are prioritized in the list and that the upsell string only appears when the user has no third party providers.

## Privacy
  1.	Does this change involve the collection, use, or sharing of new Personal Data? As far as I can tell, no.
  2.	Does this change involve a new or changed use or sharing of existing Personal Data? As far as I can tell, no.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by annaxuphoto** [comment]
I don't see any issues with the code itself, just a couple of things:
- can you provide more detailed examples/image or screen recording of the redirect change?
- I added @dju90 to approve the changes from a product perspective
Thanks!

**Comment by Hamms** [comment]
FYI @annaxuphoto that PRs from external contributors don't automatically run CI tests; you'll want to open a separate PR to do that

**Comment by annaxuphoto** [comment]
Gotcha, I'll push a separate branch to trigger a run

File: apps/src/templates/teacherDashboard/teacherSectionsRedux.js [hunk 1]
```diff
@@ -297,6 +298,13 @@ export const finishEditingSection = () => (dispatch, getState) => {
           serverSection: result
         });
         resolve(result);
+        if (section.id < 0) {
```
Comment (annaxuphoto) [inline apps/src/templates/teacherDashboard/teacherSectionsRedux.js]: nit: can we add some comments here to explain the redirect?

**Comment by annaxuphoto** [comment]
Looks like this change causes some UI tests to fail https://drone.cdn-code.org/code-dot-org/code-dot-org/5742/2/4

Let me know if you have trouble accessing the Drone build or the error logs

**Pull request closed without merging**